src/main/java/models/Game.java

```diff
@@ -10,13 +10,20 @@
 public class Game {
 
     public int score;
+    public Deck d;
 
+/*
     public java.util.List<Card> deck = new ArrayList<>();
+*/
 
     public java.util.List<java.util.List<Card>> cols = new ArrayList<>();
 
 
-    public Game(){
+    public Game(int deckType){
+
+        if (deckType == 0) d = new UDeck();
+        else d = new SDeck();
```

Comment:
Normally this would work, but it seems the way that the game object is passed to and from the model messes it up. Perhaps make two objects of type UDeck and SDeck?